### ISSUE TITLE
`:doc` output should list the interface's constraints

### ISSUE THREAD

**Issue by pyrtsa** [idris-lang/Idris-dev#3157]
This seems like an obvious thing: the `:doc` documentation of an interface should list the interface's constraints, just like it does list all child interfaces.

For example, we have `interface Eq ty => Ord ty where ...`, but when I ask the REPL, I can see no mention of `Eq` anywhere:

```
Idris> :doc Ord
Interface Ord
    The Ord interface defines comparison operations on ordered data types.

Parameters:
    ty

Methods:
    compare : Ord ty => ty -> ty -> Ordering
        
        
        The function is Total
    (<) : Ord ty => ty -> ty -> Bool
        
        infixl 6
        
        The function is Total
    (>) : Ord ty => ty -> ty -> Bool
        
        infixl 6
        
        The function is Total
    (<=) : Ord ty => ty -> ty -> Bool
        
        infixl 6
        
        The function is Total
    (>=) : Ord ty => ty -> ty -> Bool
        
        infixl 6
        
        The function is Total
    max : Ord ty => ty -> ty -> ty
        
        
        The function is Total
    min : Ord ty => ty -> ty -> ty
        
        
        The function is Total
Implementations:
    Ord ()
    Ord Int
    Ord Integer
    Ord Double
    Ord Char
    Ord String
    Ord Bool
    Ord a => Ord b => Ord (a, b)
    Ord Bits8
    Ord Bits16
    Ord Bits32
    Ord Bits64
    Ord Nat
    Ord a => Ord (List a)
    Ord Prec
    Ord (Fin n)
    Ord a => Ord (Vect n a)

Child interfaces:
    MaxBound ty
    MinBound ty
```

(The example was produced on Idris 0.11.1.)

**Comment by david-christiansen** [comment]
Thanks for a detailed and well-written report!

**Comment by owickstrom** [comment]
I had a quick look at this, thought it might be something I could contribute as a beginner in Idris. I've just started poking around in the compiler.

Anyway, I found the `InterfaceInfo` type (https://github.com/idris-lang/Idris-dev/blob/master/src/Idris/AbsSyntaxTree.hs#L1397-L1406) which is used by the `Docs` module to pretty print docs. I also found the `PInterface` constructor (https://github.com/idris-lang/Idris-dev/blob/master/src/Idris/AbsSyntaxTree.hs#L714-L723) which to my understanding is the "earlier" representation. It seems the interface parameter constraints are not present in `InterfaceInfo`, which I guess they need to be in order to print them in `Docs`. Would it be reasonable to pass along a `[(Name, Name)]`, with parameter names and constraint names, to the `InterfaceInfo` constructor? And if so, where should this conversion be done? I can't find where the `InterfaceInfo` value is created.

Feedback is very welcome. :)
